editor/components/inserter/menu.js

```diff
@@ -2,20 +2,83 @@
  * External dependencies
  */
 import { connect } from 'react-redux';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/filter';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 import Dashicon from 'components/dashicon';
+import { focusNextFocusableRef, focusPreviousFocusableRef, listVisibleBlocksByCategory } from './helpers';
 
 class InserterMenu extends wp.element.Component {
 	constructor() {
 		super( ...arguments );
+		this.blockTypes = wp.blocks.getBlocks();
+		this.categories = wp.blocks.getCategories();
+		this.nodes = {};
 		this.state = {
-			filterValue: ''
+			filterValue: '',
+			focusedElementRef: null
 		};
 		this.filter = this.filter.bind( this );
+		this.isShownBlock = this.isShownBlock.bind( this );
+		this.changeMenuSelection = this.changeMenuSelection.bind( this );
+		this.setSearchFocus = this.setSearchFocus.bind( this );
+		this.bindReferenceNode = this.bindReferenceNode.bind( this );
+	}
+
+	bindReferenceNode( nodeName ) {
+		const binder = ( node ) => this.nodes[ nodeName ] = node;
+		binder.bind( this );
+
+		return binder;
+	}
+
+	componentDidMount() {
+		// Build a stream of keydown events on component mount.
+		this.keyStream$ = Observable.fromEvent( document, 'keydown' );
```

Comment:
> I don't want to be that person that's always reluctant to changes

I think everyone is like that from time to time. It is good in my opinion to be reluctant, maybe we will decide that RxJS is not the right tool for this project, and it will be for the better.